brainiak/reconstruct/iem.py

```diff
@@ -0,0 +1,290 @@
+        n_train = len(y)
+        F = np.empty((n_train, self.n_channels))
+        for i_tr in range(n_train):
+            # Find channel activation for this orientation
+            k_min = np.argmin((y[i_tr] - self.C_D_)**2)
+            F[i_tr, :] = self.C_[:, k_min]
+        clf = linear_model.LinearRegression(fit_intercept=False,
+                                            normalize=False)
+        clf.fit(F, X)
+        self.W_ = clf
+        return self
+
+    def predict(self, X):
+        """Use test data to predict feature
+
+        Parameters
+        ----------
+            X: numpy matrix of voxel activation from test trials
+            [observations, voxels]. Used to predict feature
+            associated with the given observation.
+
+        Returns
+        -------
+            pred_dir: numpy array of estimated feature values.
+        """
+        # Check that there is enough data.. should be more
+        # samples than voxels (i.e. X should be tall)
+        shape_data = np.shape(X)
+        if len(shape_data) < 2:
+            raise ValueError("Not enough data")
+        else:
+            if np.size(X, 0) <= np.size(X, 1):
+                raise ValueError("Data Matrix ill-conditioned")
+        pred_response = self._predict_directions(X)
+        pred_indx = np.argmax(pred_response, axis=1)
+        pred_dir = self.C_D_[pred_indx]
+        return pred_dir
+
+    def score(self, X, y):
+        """Calculate error measure of prediction.
+
+        Parameters
+        ----------
+            X: numpy matrix of voxel activation from new data
+                [observations,voxels]
+            y: numpy array of responses. [observations]
+
+        Returns
+        -------
+            rss: residual sum of squares of predicted
+                features compared to to actual features.
+        """
+        pred_dir = self.predict(X)
+        u = ((y - pred_dir)**2).sum()
+        v = ((y - np.mean(y))**2).sum()
+        rss = (1 - u/v)
+        return rss
+
+    def get_params(self, deep=True):
+        """Returns model parameters.
+
+        Parameters
+        ----------
+        deep: boolean. default true.
+            if true, returns params of sub-objects
+
+        Returns
+        -------
+        params: parameter of this object
+        """
+        return{"n_channels": self.n_channels,
+               "range_start": self.range_start,
+               "range_stop": self.range_stop}
+
+    def set_params(self, **parameters):
+        """Sets model parameters after initialization.
+
+        Parameters
+        ----------
+            params: structure with parameters and change values
+
+        Returns
+        -------
+            self.
+        """
+        for parameter, value in parameters.items():
+            setattr(self, parameter, value)
+        return self
+
+    def _define_channels(self):
+        """Define basis functions (aka channels).
+
+        Parameters
+        ----------
+            self.
```

Comment:
Do we need to list `self` in the list of input params?

Reply:
We should not include `self`. See the following guideline referenced in our contributing docs:
"Do not include the `self` parameter in the ``Parameters`` section."
http://www.sphinx-doc.org/en/stable/ext/example_numpy.html

Reply:
got rid of any of those.